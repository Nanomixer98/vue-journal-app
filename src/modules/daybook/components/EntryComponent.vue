<template>
  <div
    class="entry-container b-3 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: 23 } })"
  >
    <!-- Titulo -->
    <div class="entry-tile d-flex">
      <span class="text-success fs-5 fw-bold">19</span>
      <span class="mx-1 fs-5">Agosto</span>
      <span class="mx-2 fw-light">2022, sabado</span>
    </div>

    <div class="entry-description">
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptatibus
      quos a perferendis possimus. Distinctio laudantium molestiae quod cum
      dolore asperiores pariatur doloribus consectetur, odio natus explicabo
      dicta. Quae, ducimus itaque.
    </div>
  </div>
</template>

<script>
export default {};
</script>

<style lang="scss">
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in;
  }
  .entry-description {
    font-size: 12px;
  }
}
</style>
