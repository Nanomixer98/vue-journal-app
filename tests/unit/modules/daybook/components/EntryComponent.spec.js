import { shallowMount } from "@vue/test-utils";
import EntryComponent from '@/modules/daybook/components/EntryComponent'
import { journalState } from "@/../tests/unit/mock-data/test-journal-state";

describe('Pruebas en Entry Component', () => {
    const mockRouter = {
        push: jest.fn()
    }
    let wrapper = shallowMount(EntryComponent, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    });

    it('debe de hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('debe de redireccionar al hacer click en el entry-container', () => {
        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')

        expect(mockRouter.push).toHaveBeenCalledWith({ 
            "name": "entry", 
            "params": { 
                "id": journalState.entries[0].id
            } 
        })
    })

    it('pruebas en las propiedades computadas', () => {
        expect(wrapper.vm.day).toBe(12);
        expect(wrapper.vm.month).toBe('Septiembre');
        expect(wrapper.vm.yearDay).toBe('2022, Lunes');
    })

})