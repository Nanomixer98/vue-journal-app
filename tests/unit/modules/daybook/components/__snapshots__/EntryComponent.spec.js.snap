// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Entry Component debe de hacer match con el snapshot 1`] = `
<div class="entry-container b-3 pointer p-2">
  <!-- Titulo -->
  <div class="entry-tile d-flex"><span class="text-success fs-5 fw-bold">12</span><span class="mx-1 fs-5">Septiembre</span><span class="mx-2 fw-light">2022, Lunes</span></div>
  <div class="entry-description">IT'S ALIVE 2</div>
</div>
`;
